src/lightning/pytorch/callbacks/model_checkpoint.py

```diff
@@ -240,6 +241,7 @@ def __init__(
         self.best_model_score: Optional[Tensor] = None
         self.best_model_path = ""
         self.last_model_path = ""
+        self._last_checkpoint_saved = ""
```

Comment:
`last_model_path` is the path to the "last.ckpt" file, whereas `self._last_checkpoint_saved` points to the latest checkpoint saved. They can't be unified because they aren't the same thing. This PR makes it so that the file saved at  `self.last_model_path` is a symlink to the file at `self._last_checkpoint_saved`